Scripts/script-ProofpointDecodeURL.yml

```diff
@@ -13,23 +13,26 @@ script: |
   valid_versions = ['v1', 'v2']
 
   def find_version(url):
-       match = re.search(r'https://urldefense.proofpoint.com/(?P<version>v[0-9])/', url)
-       return match.group('version')
+      match = re.search(r'https://urldefense.proofpoint.com/(?P<version>v[0-9])/', url)
+      return match.group('version')
 
   def ppdecode(url, return_all=False):
       try:
           version = find_version(url)
       except AttributeError:
-          raise RuntimeError('This does not appear to be a valid proofpoint url: {}'.format(url))
+          return_error('This does not appear to be a valid proofpoint url: {}'.format(url))
       else:
           if version not in valid_versions:
-              raise RuntimeError('Ppdecode is unprepared to handle this version of proofpoint urls: {}'.format(url))
+              return_error('Ppdecode is unprepared to handle this version of proofpoint urls: {}'.format(url))
       parsed_url = urlparse(url)
       query_components = parse_qs(parsed_url.query)
-      if sys.version_info[0] < 3:
-          translated_url = query_components['u'][0].translate(string.maketrans('-_', '%/'))
+      if 'u' in query_components:
+          if sys.version_info[0] < 3:
+              translated_url = query_components['u'][0].translate(string.maketrans('-_', '%/'))
+          else:
+              translated_url = query_components['u'][0].translate(str.maketrans('-_', '%/'))
       else:
-          translated_url = query_components['u'][0].translate(str.maketrans('-_', '%/'))
+          return_error('A URL was not detected in the query.')
```

Comment:
This means that if previously we returned something - now we'll fail the script, correct? Is this intentional?

Reply:
I think once we failed on unexpected error, and now it will fail with nicer message - this is the whole issue is about

Reply:
It's intentional. The original issue was that when it failed, it didn't give the user a good indication as to why.

The decode url function looks for the query string in the url and then uri-decodes it. When there is no query string, it fails. This was a way to make it more of a "The query string you provided has no url in it" versus "The script just failed on a key error"

Reply:
try to use the script with bad url (without u) and use extend-context to set the error in the context, this way you will be able to check the error.

Reply:
Just checked this. The script itself is using a regex to grab the url (specifically looking for the "u") and then decodes the query string. Since this is all done on our side and doesn't hit proofpoint at all, there's no context to extend.